torch/_decomp/decompositions.py

```diff
@@ -5036,6 +5036,62 @@ def scaled_dot_product_flash_attention_for_cpu(
     return output, attn
 
 
+@register_decomposition(aten._scaled_dot_product_int8_for_cpu.default)
+def scaled_dot_product_int8_for_cpu(
+    query: Tensor,
+    key: Tensor,
+    value: Tensor,
+    dropout_p: float = 0.0,
+    is_causal: bool = False,
+    *,
+    attn_mask: Optional[Tensor] = None,
+    scale: Optional[float] = None,
+    q_zp: int = 0,
+    q_scale: float = 0.0,
+    k_zp: int = 0,
+    k_scale: float = 0.0,
+    v_zp: int = 0,
+    v_scale: float = 0.0,
+    a_zp: int = 0,
+    a_scale: float = 0.0,
+    o_zp: int = 0,
+    o_scale: float = 0.0,
+) -> Tensor:
+    dtype = query.dtype
+    torch._check(
+        dtype == torch.uint8,
+        lambda: f"query must be U8 but got {dtype}",
+    )
+    torch._check(
+        query.dim() == 4 and key.dim() == 4 and value.dim() == 4,
+        lambda: f"q, k, v must be a 4 dimensional tensor, got {query.dim()}, {key.dim()}, {value.dim()}",
+    )
+    torch._check(
+        dropout_p == 0.0, lambda: f"dropout probability must be zero, got {dropout_p}"
+    )
+    torch._check(
+        query.shape[3] == value.shape[3] and key.shape[3] == value.shape[3],
+        lambda: "q, k, v should have the same head size",
+    )
+
+    output = aten._scaled_dot_product_int8_for_cpu.default(
```

Comment:
Suggest you add a note to link to that in `scaled_dot_product_flash_attention_for_cpu`, explaining why we need a decomposition that calls to the function itself (which sounds really tricky to me).

Reply:
Yes, you are right. The decomposition is removed.